src/main/java/alex/mojaki/s3upload/StreamTransferManager.java

```diff
@@ -470,6 +517,9 @@ private void uploadStreamPart(StreamPart part) {
                 .withUploadId(uploadId).withPartNumber(part.getPartNumber())
                 .withInputStream(part.getInputStream())
                 .withPartSize(part.size());
+        if (checkIntegrity) {
+            uploadRequest.withMD5Digest(part.getMD5Digest());
```

Comment:
if you are not chaining, you could probably use `setMD5Digest` instead - `with` sometimes is immutable

Reply:
done